core/trino-main/src/main/java/io/trino/operator/LeafTableFunctionOperator.java

```diff
@@ -167,6 +168,9 @@ public Page getOutput()
                 return null;
             }
             currentSplit = pendingSplits.remove(0);
+            if (currentSplit instanceof EmptySplit) {
+                return null;
```

Comment:
`EmptySplit` is only instantiated in `SourcePartitionedScheduler`. However, I find it difficult to understand on what conditions it is created:

- is it created only when the `splitSource` is empty? Can it be created when the `splitSource` provided some splits, or only in the case when there were no splits at all?
- can there be more than one `EmptySplit` created per splitSource?

I'd like to understand before I confirm that the proposed solution is right.
@sopel39 could you please help?

Reply:
> is it created only when the splitSource is empty? Can it be created when the splitSource provided some splits, or only in the case when there were no splits at all?

Only if there were no actual splits generated IIRC. However, I wouldn't rely on that. Nothing should break if `EmptySplit` is generated always IMO.

> can there be more than one EmptySplit created per splitSource?

I don't think so. Again, I wouldn't rely that there will be at most one `EmptySplit`.

Reply:
@sopel39 thanks for the explanation.

@homar considering the above, I think that the safe and future-proof way to handle `EmptySplit` is to iterate over `pendingSplits` dropping `EmptySplit`s until there are no more splits on the queue, or we found a valid split.

Reply:
ok, thank you